docs/generating-tests.rst

```diff
@@ -0,0 +1,370 @@
+
+.. note::
+   ``testeth`` cannot understand LLL if the system does not have the LLL compiler installed. The LLL compiler is currently distributed as part of the `Solidity`_ compiler.
+
+
+Writing or modifying a Test Filler
+----------------------------------
+
+A test filler file should always correspond to one test case, so a single GeneralStateTest filler file is not supposed to contain multiple tests.  ``testeth`` tool still accepts multiple GeneralStateTest fillers in a single test filler file, but this might change.
+
+In the ``tests`` repository, the test filler files for GeneralStateTests live under ``src/GeneralStateTestsFiller`` directory. The directory has many subdirectories. You need to choose one of the subdirectories or create one.  The name of the filler file needs to end with ``Filler.json``.  For example, we might want to create a new directory ``src/GeneralStateTestsFiller/stExample2`` with a new filler file ``returndatacopy_initialFiller.json``, or edit one of the existing filler files in the directory structure.
+
+.. note::
+   If you create a new directory here, you need to add one line in ``Aleth`` and file that change in a pull request to ``Aleth``.
+
+For creating a new test filler, the easiest way to start is to copy an existing filler file. The first thing to change  is the name of the test in the beginning of the file. The name of the test should coincide with the file name except ``Filler.json`` [#]_. For example, in the file we created above, the filler file contains the name of the test ``returndatacopy_initial``.  The overall structure of ``returndatacopy_initialFiller.json`` should be:
+
+.. code::
+
+   {
+       "returndatacopy_initial" : {
+          "env" : { ... }
+          "expect" : [ ... ]
+          "pre" " { ... }
+          "transaction" : { ... }
+       }
+   }
+
+
+where ``...`` indicates omissions.
+
+.. [#] The file name and the name written in JSON should match because ``testeth`` prints the name written in JSON, but the user needs to find a file.
+
+
+``env`` field contains some parameters in a straightforward way.
+
+``pre`` field describes the pre-state account-wise:
+
+.. code::
+
+     "pre" : {
+        "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6" : {
+            "balance" : "0x0de0b6b3a7640000",
+            "code" : "{ (MSTORE 0 0x112233445566778899aabbccddeeff) (RETURNDATACOPY 0 0 32) (SSTORE 0 (MLOAD 0)) }",
+            "code" : "0x306000526020600060003e600051600055",
+            "nonce" : "0x00",
+            "storage" : {
+                "0x00" : "0x01"
+            }
+        }
+     }
+
+
+As specified in the Yellow Paper, an account contains a balance, a code, a nonce and a storage.
+
+Notice the ``code`` field is duplicated.  If many fields exist under the same name, the last one is used. In this particular case, the LLL compiler was not ready to parse the new instruction ``RETURNDATACOPY`` so a compiled runtime bytecode is added as the second ``code`` field [#]_.
```

Comment:
duplicates were used as a comments. now to make a comment in json any field starting with `//` is threated as comment.

Reply:
Ok, I've added an example comment to the source extract and adopted the text accordingly.